app/src/main/scala/org/alephium/explorer/api/model/StdInterfaceId.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.api.model
+
+import scala.collection.immutable.ArraySeq
+
+import sttp.tapir.{Schema, Validator}
+import upickle.core.Abort
+
+import org.alephium.json.Json._
+
+sealed trait StdInterfaceId {
+  def value: String
+  def id: Int
+}
+
+object StdInterfaceId {
+
+  case object FungibleToken extends StdInterfaceId {
+    val value: String = "fungible"
+    val id: Int       = 1
+  }
+
+  case object NFTCollection extends StdInterfaceId {
+    val value: String = "non-fungible-collection"
+    val id: Int       = 2
+  }
+
+  case object NFT extends StdInterfaceId {
+    val value: String = "non-fungible"
+    val id: Int       = 3
+  }
+
+  case object NonStandard extends StdInterfaceId {
+    val value: String = "non-standard"
+    val id: Int       = 0
+  }
+
+  def from(code: String): StdInterfaceId =
+    code match {
+      case "0001" => FungibleToken
+      case "0002" => NFTCollection
+      case "0003" => NFT
+      case _      => NonStandard
+    }
```

Comment:
We have made some changes to std interface ids too: https://github.com/alephium/alephium-web3/blob/master/packages/web3/src/api/types.ts#L286-L291

One interface inheritates another if its id is a decent of another's.

Reply:
:+1: it's updated